### ISSUE TITLE
Scheduler and autoCommit connections

### ISSUE THREAD

**Issue by greyridgesupport** [kagkarlsson/db-scheduler#97]
As far as I can work out, db-scheduler does not work unless autoCommit (the worst idea in the history of database connectivity) is set to true.  I've tried the following code (kotlin but would be the same, except longer, in java I think)

```kotlin
fun main(){
  val config = HikariConfig()
  config.jdbcUrl = "jdbc:oracle:thin:test/test@localhost:1521:test"
  config.isAutoCommit = false

  val datasource = HikariDataSource(config)

  val hourlyTask: RecurringTask<Void> = Tasks.recurring("my-hourly-task", FixedDelay.ofHours(1))
    .execute({ inst, ctx -> println("Executed!") })

  val scheduler: Scheduler = Scheduler
    .create(datasource)
    .startTasks(hourlyTask)
    .threads(5)
    .build()

  datasource.connection.commit()

  scheduler.start()

  datasource.connection.commit()
}
```

and despite adding commit everywhere I can think of, it doesn't work.  If I take out the commits and change config.autoCommit to true, it works as expected.

Is it possible to use db-scheduler with datasources which are not autocommit or am I doing something daft (more than possible!)

**Comment by kagkarlsson** [comment]
Yes I think you are correct, db-scheduler assumes `autocommit` is true. If the connection pool has set it to `false` before handing the connection to db-scheduler I suppose nothing will ever be committed. 

Didn't know setting it to `false` was a thing 😬. Sounds like we need to add some handling for that case.

**Comment by kagkarlsson** [comment]
I usually use a Spring transaction manager which disables autocommit on the connections

**Comment by kagkarlsson** [comment]
Just to let you know, I have a fix for this coming up 👍

**Comment by kagkarlsson** [comment]
Fixed in v7.1
